tests/Exizent.CaseManagement.Client.IntegrationTests/UpdatingEstateItemNotes.cs

```diff
@@ -0,0 +1,233 @@
+                settings.Scope = ExizentScopes.All;
+                settings.UserAgent = "My browser";
+            }
+        );
+
+        await using var serviceProvider = serviceContainer.BuildServiceProvider();
+        using var scope = serviceProvider.CreateScope();
+        var client = scope.ServiceProvider.GetRequiredService<ICaseManagementApiClient>();
+
+        await client.UpdateEstateItemNotes(caseId, estateItemId, "some notes");
+    }
+
+    [Fact]
+    public async Task ShouldErrorWhenUpdateEstateItemNotesFailsWith404NotFound()
+    {
+        var clientId = Guid.NewGuid().ToString();
+        var clientSecret = Guid.NewGuid().ToString();
+
+        var caseId = Guid.NewGuid();
+        var estateItemId = Guid.NewGuid();
+
+        using var authApiServer = WireMockServer.Start();
+        authApiServer
+            .Given(Request.Create().WithPath("/oauth/token")
+                .WithBody(new JsonPartialMatcher(
+                    $@"{{ ""grant_type"": ""client_credentials"", ""client_id"": ""{clientId}"", ""client_secret"": ""{clientSecret}"", ""scope"": ""{ExizentScopes.All}"", ""audience"": ""https://resources.exizent.com"" }}"))
+                .UsingPost())
+            .RespondWith(Response.Create()
+                .WithBody("{ \"access_token\": \"123456\", \"expires_in\": 3600, \"token_type\": \"Bearer\" }")
+            );
+
+        using var casesApiServer = WireMockServer.Start();
+        casesApiServer.Given(
+                Request.Create().WithPath($"/cases/{caseId}/estateitems/{estateItemId}/notes").UsingPut()
+                    .WithHeader("User-Agent", "My browser")
+            )
+            .RespondWith(Response.Create()
+                .WithHeader("Authorization", "Bearer 123456")
+                .WithStatusCode(HttpStatusCode.NotFound)
+            );
+
+        var baseUri = casesApiServer.Url;
+        var baseAuthorizationUri = authApiServer.Url;
+
+        var serviceContainer = new ServiceCollection();
+        serviceContainer.AddExizentCaseManagementClient(
+            clientId,
+            clientSecret,
+            settings =>
+            {
+                settings.BaseUri = new Uri(baseUri);
+                settings.BaseAuthorizationUri = new Uri(baseAuthorizationUri);
+                settings.Scope = ExizentScopes.All;
+                settings.UserAgent = "My browser";
+            }
+        );
+
+        await using var serviceProvider = serviceContainer.BuildServiceProvider();
+        using var scope = serviceProvider.CreateScope();
+        var client = scope.ServiceProvider.GetRequiredService<ICaseManagementApiClient>();
+
+        Func<Task> action = async () => await client.UpdateEstateItemNotes(caseId, estateItemId, "some notes");
+
+        await action.Should().ThrowAsync<HttpRequestException>()
+            .WithMessage("Response status code does not indicate success: 404 (Not Found).");
+    }
+
```

Comment:
what is the difference between this and the above test?

Reply:
in this test I'm setting the response to be .WithStatusCode(HttpStatusCode.NotFound), whereas in the first one I'm returning a no content (.WithStatusCode(HttpStatusCode.NoContent))

In the one below, the response is created without a body, so I mock a NotFound again